modules/kafka/src/main/scala/higherkindness/mu/format/AvroWithSchema.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017-2020 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.format
+
+import java.io.ByteArrayOutputStream
+import com.sksamuel.avro4s._
+
+object AvroWithSchema {
+  implicit def encoder[T: SchemaFor: ToRecord]: Encoder[T] = new Encoder[T] {
+    override def encode(t: T): Array[Byte] = {
+      val bOut = new ByteArrayOutputStream()
```

Comment:
this looks pretty similar to this
https://github.com/higherkindness/mu-scala/blob/136f81ea0fb2b2d1a31089895b0724fccd7a3018/modules/internal/src/main/scala/higherkindness/mu/rpc/internal/encoders/avro.scala#L58
Would it be possible to put the code in one place and reuse it?

Reply:
Yes I thought about this as well. I will add a ticket to refactor to share the common code.